symfit/core/models.py

```diff
@@ -510,9 +510,10 @@ def __init__(self, model, independent_vars=None, params=None, **kwargs):
                     sub_model[var] = expr
             if sub_model:
                 sub_model = BaseModel(sub_model)
-                # Update with the users input. In case of conflict, this prioritizes
-                # the info given by the user.
-                connectivity_mapping.update(sub_model.connectivity_mapping)
+                # Update with the users input. In case of conflict, this
+                # prioritizes the info given by the user.
+                sub_model.connectivity_mapping.update(connectivity_mapping)
+                connectivity_mapping = sub_model.connectivity_mapping
 
             self.connectivity_mapping = connectivity_mapping
```

Comment:
Now you get a funny case where sometimes (no analytical component) `self.connectivity_mapping` is an alias to the user input, and other times it isn't. So maybe add a `.copy()`. And yes, I'm splitting hairs here.

Reply:
Yes, but it's a good remark because that is indeed a bit inconsistent. I can definitely see @Jhsmit come up with some way to complain about this in the future.